pilot/pkg/model/sidecar.go

```diff
@@ -266,7 +309,15 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *Config, configNamespa
 	// that these services need
 	out.destinationRules = make(map[host.Name]*Config)
 	for _, s := range out.services {
-		out.destinationRules[s.Hostname] = ps.DestinationRule(&dummyNode, s)
+		dr := ps.DestinationRule(&dummyNode, s)
+		if dr != nil {
+			out.destinationRules[s.Hostname] = dr
+			out.AddConfigDependencies(ConfigKey{
+				Kind:      DestinationRuleKind,
+				Name:      dr.Name,
```

Comment:
This dr is an combineDestinationRule, the name is meaningless and cannot be used as a key for dependence

Reply:
The cases in which config name is meaningless will be left to later optimization.